src/com/unbounce/treajure/map.clj

```diff
@@ -6,3 +6,24 @@
   (if (every? map? vals)
     (apply merge-with deep-merge vals)
     (last vals)))
+
+(defn update-if-exists
+  "Variation of clojure.core/update-in which performs no-operation if the
+  key is not present in the map. "
+  [m keys f & args]
+  (let [value (get-in m keys ::not-found)]
+    (if-not (identical? value ::not-found)
+      (assoc-in m keys (apply f value args))
+      m)))
+
+(defn map-values
+  "Create a new map of same keys and applying f to all the values.
+
+  Taken from plumatic/plumbing.
+  "
+  [m f]
```

Comment:
y'all lost me at transducers. Admittedly multi-arity functions is not something that's an "obvious" thing for me to reach for - but `f` being just 1-arity seems right. Incidentally, `MapLike` things in Scala have the same `mapValues` function where `f:(V) => W` 😛